src/test/java/com/direwolf20/buildinggadgets/common/building/region/RegionIteratorTest.java

```diff
@@ -0,0 +1,50 @@
+package com.direwolf20.buildinggadgets.common.building.region;
+
+import com.direwolf20.buildinggadgets.common.building.Region;
+import com.google.common.collect.ImmutableList;
+import net.minecraft.util.math.BlockPos;
+import org.junit.jupiter.api.*;
+
+import java.util.Iterator;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RegionIteratorTest {
+
+    @Test
+    public void iteratorShouldHaveSameNumberOfElementsAsSizeMethod() {
+        Region region = new Region(1, 1, 1, 3, 3, 3);
+        assertEquals(region.size(), ImmutableList.copyOf(region.iterator()).size());
+    }
+
+    @Test
+    public void iteratorShouldStartGoInAccentingOrderWhereZFirstYSecondXThird() {
+        Region region = new Region(1, 1, 1, 2, 2, 2);
+        Iterator<BlockPos> it = region.iterator();
+
+        assertEquals(new BlockPos(1, 1, 1), it.next());
+        assertEquals(new BlockPos(1, 1, 2), it.next());
+        assertEquals(new BlockPos(1, 2, 1), it.next());
+        assertEquals(new BlockPos(1, 2, 2), it.next());
+        assertEquals(new BlockPos(2, 1, 1), it.next());
+        assertEquals(new BlockPos(2, 1, 2), it.next());
+        assertEquals(new BlockPos(2, 2, 1), it.next());
+        assertEquals(new BlockPos(2, 2, 2), it.next());
```

Comment:
You can avoid hardcoding this, by collecting the iterator into an List, and then verifying that the List is sorted according to the usual Comparator.